client/src/main/proto/spine/validation.proto

```diff
@@ -57,43 +56,173 @@ extend google.protobuf.FieldOptions {
     ValidOption valid = 51009;
 }
 
-//TODO:2016-02-16:alexander.yevsyukov: Document each message option.
-
-//TODO:2016-02-16:alexander.yevsyukov: Take default messages from Hibernate Validator.
-//TODO:2016-02-16:alexander.yevsyukov: Have default texts formatted messages and document message format parameters as comment for `option (default_message)`.
+// TODO:2016-03-25:alexander.litus: error messages format
 
+// If the field type is `message`, it must be set to a non-default instance;
+// if it is `string` or `bytes`, the value must not be an empty string or array.
+// Other field types are not applicable.
+// If the field is repeated, it must have at least one value set, and all its values must be valid.
+//
+// Example:
+//
+// MyMessage field = 1 [(required).is = true];
+//
 message RequiredOption {
-    // The default format string for missing field. The parameter is the name of the missing field.
-    option (default_message) = "Field %s must be set";
+    // The default error message format string.
+    // The format parameter is the simple name of the missing field.
+    option (default_message) = "'%s' must be set";
 
-    bool value = 1;
+    // Specifies if the field is required. Must be set to true. Otherwise, the field is not validated.
+    // The default value is false.
+    bool is = 1;
 
-    // User-defined validation error message format.
+    // A user-defined validation error format message.
+    // The format parameters are the same as for the default message.
     string msg = 2;
 }
 
+// TODO:2016-03-14:alexander.litus: is this option really needed? Why not use min/max instead?
+// Applicable only to numbers.
 message DecimalOption {
-    option (default_message) = "Field value must fit into range";
+    option (default_message) = "field value must fit into range";
```

Comment:
Have messages as sentences as we discussed. We probably need to have the range mentioned in the default message format.